src/NewChain/AuxiliaryChain.ts

```diff
@@ -274,12 +275,80 @@ export default class AuxiliaryChain {
     node.startSealer(gasPrice, targetGasLimit, bootKeyFile);
     // The sealer runs locally on this machine and the port is published to the host from the
     // docker container.
-    this.logInfo('waiting 5 seconds for the sealer port to become available');
-    await AuxiliaryChain.sleep(5000);
     // Has to be RPC and not WS. WS connection was closed before deploying the Co-Gateway.
     // Reason unknown. Possibly due to the fact that according to `lsof` node keeps opening new
     // connections.
     this.web3 = new Web3(`http://127.0.0.1:${this.nodeDescription.rpcPort}`);
+    await this.verifyAccountsUnlocking();
+  }
+
+  /**
+   * It polls every 4-secs to fetch the list of wallets.
+   * It logs error when connection is not established even after max tries.
+   */
+  private async verifyAccountsUnlocking() {
+    let noOfRetries;
+    let unlockStatus;
+    for (noOfRetries = 1; noOfRetries <= this.maxRetries; noOfRetries++) {
+      this.logInfo(`number of tries to fetch unlocked accounts is ${noOfRetries}`);
+      let response;
+      await AuxiliaryChain.sleep(4000);
+      try {
+        response = await this.getUnlockedAccounts();
+        if (response.result.length > 0) {
+          unlockStatus = this.getAccountsStatus(response.result);
+          if (unlockStatus) {
+            this.logInfo('accounts unlocked');
+            break;
```

Comment:
1. Very deeply nested `for`-loop. Not very readable.
2. `break` *should* be avoided. Can be ok here, of course, but have you thought about other designs that may not require a `break`? Especially a `break` in the success case?

For example, instead of the deep `for`-loop, maybe you could have a (much flatter) loop like
```js
const nrOfTries = 10;
const sleepTimeInSeconds = this.timeToWaitForAccountsToUnlockInSeconds / nrOfTries;
let totalWaitTimeInSeconds = 0;
while (!this.accountsUnlocked()) {
  await AuxiliaryChain.sleep(sleepTimeInSeconds * 1000);
  totalWaitTimeInSeconds += sleepTimeInSeconds;
  if (totalWaitTimeInSeconds > this.timeToWaitForAccountsToUnlockInSeconds) {
    throw new Error('node did not unlock accounts in time');
  }
}
```

And then inside this function the main code would happen:
```ts
private function accountsUnlocked(): boolean {
  // This is renamed from `this.getUnlockedAccounts()` which was a misleading name:
  const wallets = this.getWallets();
  const accountsUnlocked = this.walletsUnlocked(wallets);

  return accountsUnlocked;
}
```

And inside `function walletsUnlocked(wallets): boolean` you would do your check that the wallets are unlocked.

Reply:
Thank you for the suggestion. I liked the suggested approach as it helped to remove `break` statement.
I have diverted with the approach slightly to calculate `sleepTimeInSeconds`( I have used variable name `timeToWaitInSecs`). Please see the approach taken and let me know your thoughts.